NAMING_CONVENTIONS.md

```diff
@@ -0,0 +1,37 @@
+Arnold USD naming conventions
+=======================
+
+The Arnold USD implementation supports arnold-specific attributes through custom schemas.
+
+## Arnold attributes on USD nodes
+
+Additional arnold attributes on USD-native nodes (geometries, lights, etc...) need to be in the "arnold" namespace
+
+Example:
+def DomeLight "my_light"
+{
+ float intensity=2
+ int arnold:max_bounces=99
+}
+
+## Arnold Shaders
+
+Arnold shaders are supported with UsdShade Shader primitives. The info:id attribute should be the shader name, with an "arnold" namespace. The attributes will be in the "inputs" scope.
+
+Example:
+def Shader "my_shader"
+{
+    uniform token info:id = "arnold:standard_surface"
+    color3f inputs:base_color = (0,0,1)
+}
+
+## Arnold nodes
+
+Any Arnold node can be represented in USD, with a camel-cased node type, prefixed with "Arnold". 
```

Comment:
Can you explain why we need to transliterate Arnold node names to CamelCase (the Arnold node names are normally snake_case)? This makes it more complicated to write translation tools. What are the rules when converting from snake_case to CamelCase, especially when dealing with capitals and acronyms?

Reply:
Interesting point. FYI, every USD Schemas I saw were using camel case style.
If Arnold ones keep the snake case style, it could just look a little bit strange when looking at a usda file. Currently a Skydome prim looks like that:
`def ArnoldSkydomeLight "domeLight1" {}`
but would like a little but "use alien" like that:
`def arnold_skydome_light "domeLight1" {}`

Not a big deal for sure, but I would vote for keeping consistency with USD current Hydra plugin echo system and so use camel case.

Reply:
I just did a test, and it's not possible to build a schema class called `arnold_skydome_light`. It fails at linking time, because it needs to start with a capital. It does work with `Arnold_skydome_light`.

Reply:
I would keep it without underscores. It does not makes sense to me to try to look "almost" like native name but not really. Converting from snake_case to UpperCamelCase should not be a big issue on the tool translation side.